compiler/ballerina-lang/src/main/java/org/ballerinalang/model/clauses/GroupingKeyNode.java

```diff
@@ -0,0 +1,32 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.model.clauses;
+
+import org.ballerinalang.model.tree.Node;
+
+/**
+ * The interface with the APIs to implement the variable definition in "group by" clause.
+ *
+ * @since 2201.4.0
+ */
+public interface GroupingKeyNode extends Node {
+    void setGroupingKey(Node groupingKey);
+
+    Node getGroupingKey();
```

Comment:
May not need these depending on how we differentiate the two kinds of grouping keys. See the comments for `BLangGroupingKey`.